tests/TenantDatabaseManagerTest.php

```diff
@@ -390,6 +390,81 @@
     expect(file_exists($customPath . '/' . $name))->toBeTrue();
 });
 
+test('you can define the connection template array in template_tenant_connection or use a name of an existing connection template', function () {
+    Event::listen(TenantCreated::class, JobPipeline::make([CreateDatabase::class])->send(function (TenantCreated $event) {
+        return $event->tenant;
+    })->toListener());
+
+    config([
+        'tenancy.database.managers.mysql' => MySQLDatabaseManager::class,
+        'tenancy.database.template_tenant_connection' => 'mysql',
+    ]);
+
+    $name = 'foo' . Str::random(8);
+    $tenant = Tenant::create([
+        'tenancy_db_name' => $name,
+    ]);
+
+    /** @var MySQLDatabaseManager $manager */
+    $manager = $tenant->database()->manager();
+    expect($manager->databaseExists($name))->toBeTrue();
+    expect($manager->database()->getConfig('host'))->toBe('mysql');
+
+    config([
+        'tenancy.database.template_tenant_connection' => [
+            'driver' => 'mysql',
+            'url' => null,
+            'host' => 'mysql2',
+            'port' => '3306',
+            'database' => 'main',
+            'username' => 'root',
+            'password' => 'password',
+            'unix_socket' => '',
+            'charset' => 'utf8mb4',
+            'collation' => 'utf8mb4_unicode_ci',
+            'prefix' => '',
+            'prefix_indexes' => true,
+            'strict' => true,
+            'engine' => null,
+            'options' => [],
+        ],
+    ]);
+
+    $tenant = Tenant::create([
+        'tenancy_db_name' => $name,
+    ]);
+
+    /** @var MySQLDatabaseManager $manager */
+    $manager = $tenant->database()->manager();
+    expect($manager->databaseExists($name))->toBeTrue();
+    expect($manager->database()->getConfig('host'))->toBe('mysql2');
+});
+
+test('you can define the partial connection template array in template_tenant_connection and it will be completed by merging into the central connection', function () {
```

Comment:
I don't really like the term "partial" here, it makes things a bit less concise and harder to understand in my opinion. I'd count on people understanding`array_merge` because it's a pretty known PHP thing – the information that the template is obtained like `array_merge($central, $tenant)` (wrote concisely) is the important thing to test I think.

The main thing to improve here is the conciseness – the template is used like `array_merge($centralTemplate, $tenantTemplate)`, which the test name now does indicate, but it could be worded better.

```suggestion
test('template_tenant_connection will be merged into the central connection template using array_merge($central, $tenant)', function () {
```
My suggestion may not be the best, but I'd use something similar to it

Reply:
Partially agree with you. But the feature we want to test is `we can define partial connection config`. I don't think this name describes that.

Reply:
I see what you mean. I'll try thinking of a better name

Reply:
> the feature we want to test is we can define partial connection config

So to confirm – we want to test that if `template_tenant_connection` overrides the central connection template, and if `template_tenant_connection` is missing some things that the central template has, `template_tenant_connection` gets them?

Reply:
Btw apart from this detail, I think the PR is good

Reply:
> So to confirm – we want to test that if template_tenant_connection overrides the central connection template, and if template_tenant_connection is missing some things that the central template has, template_tenant_connection gets them?

Correct. But I'm not sure if we can say it _override_. Its actually merging `template` into `central`. Override gives a feeling of _replacing_ everything.

Reply:
Right. But I don't think I can come up with a better name than the one I suggested [here](https://github.com/archtechx/tenancy/pull/1040#discussion_r1069483913) – maybe we could even omit the `array_merge` part. @stancl, what do you think?

Reply:
Test names shouldn't include special characters or code

Reply:
Or really implementation details, just describe what is being done. We're merging the connection into the central connection. That's fine I think.

Reply:
```suggestion
test('configured tenant connection template merges into the central connection template', function () {
```

@stancl is this fine?

Reply:
`gets merged` instead of `merges`, otherwise it's okay